design/one-pager-resource-connectivity-mvp.md

```diff
@@ -0,0 +1,656 @@
+      clientKey: PEMPEMPEM
+      sslCipher: supersecure
+      verifyServerCertificate: true
+    failoverTarget: false
+```
+
+#### Google Kubernetes Engine
+
+The Crossplane `GKECluster` managed resource is particularly misleading. Its API
+definition has a fairly comprehensive (albeit confusingly flattened) facsimilie
+of the associated GKE API object, but only a small subset of these fields are
+actually parsed and submitted to the GKE API. Only the fields that are parsed
+and submitted are included in the below example:
+
+```yaml
+---
+apiVersion: compute.gcp.crossplane.io/v1alpha1
+kind: GKEInstance
+metadata:
+  namespace: crossplane-system
+  name: example
+spec:
+  nameFormat: mycoolname
+  clusterVersion: 1.13
+  createSubnetwork: false
+  enableIPAlias: true
+  labels:
+    cool: very
+  machineType: n1-standard-2
+  clusterIPV4CIDR: 192.168.0.0/16
+  nodeIPV4CIDR: 10.0.0.0/8
+  serviceIPV4CIDR: 172.16.0.0/24
+  nodeLabels:
+  - coollabel
+  numNodes: 6
+  zone: us-central1-a
+  Scopes:
+  - mycoolnodescope
+  # This setting is not currently read from the GKE resource class.
+  username: cooluser
+```
+
+By comparison, a direct translation of the [GKE cluster external resource]'s
+writable API object fields to a Kubernetes YAML specification would be as
+follows. Note that the GKE API contains several deprecated fields, all of which
+are superceded by others (e.g. `nodeConfig` is superceded by `nodePools`). The
+below translation omits these deprecated fields.
+
+```yaml
+---
+apiVersion: compute.gcp.crossplane.io/v1alpha1
+kind: GKEInstance
+metadata:
+  namespace: crossplane-system
+  name: example
+spec:
+  nameFormat: mycoolname
+  description: My cool cluster
+  masterAuth:
+    username: cooluser
+    password: secretpassword
+    clientCertificateConfig:
+      issueClientCertificate: true
+  loggingService: logging.googleapis.com/kubernetes
+  monitoringService: monitoring.googleapis.com
+  network: mycoolnetwork
+  clusterIpv4Cidr: 10.0.0.0/8
+  addonsConfig:
+    httpLoadBalancing:
+      disabled: false
+    horizontalPodAutoscaling:
+      disabled: false
+    networkPolicyConfig:
+      disabled: false
+  subnetwork: mycoolsubnetwork
+  nodePools:
+  - name: string
+    config:
+      machineType: n1-standard-2
+      diskSizeGb: 20
+      oauthScopes:
+      - cool
+      serviceAccount: mycoolserviceaccount@accounts.google.com
+      metadata:
+        cool: very
+      imageType: coreos/stable
+      labels:
+        cool: true
+      localSsdCount: 0
+      tags:
+      - cool
+      preemptible: false
+      accelerators:
+        acceleratorCount: 2
+        acceleratorType: large
+      diskType: pd-ssd
+      minCpuPlatform: Intel Haswell
+      taints:
+        - key: color
+          value: purple
+          effect: NO_SCHEDULE
+    initialNodeCount: number
+    version: 1.14
+    autoscaling:
+      enabled: true
+      minNodeCount: 3
+      maxNodeCount: 9
+    management: object (NodeManagement)
```

Comment:
This is probably a dumb question, but is `object (NodeManagement)` a label or comment type thing?

Reply:
Haha, it's an artifact of the translation process from the JSON-ish GKE API docs to YAML. It should be removed.